### PR TITLE
lib: modem_info: port to nrf_modem_at APIs

### PR DESCRIPTION
**Pull request by lemrey**
Port to `nrf_modem_at` and `at_notif`.
Escape % characters as necessary.

### CODE REVIEW

**Comment by jtguggedal** [comment]
I think @bjda will submit a significant rewrite of this library fairly soon. Perhaps we should hold off until there's a clear timeline on that effort before merging this PR

File: lib/modem_info/modem_info.c
Comment (trantanen) [inline lib/modem_info/modem_info.c:362]: Somehow I'd like our libraries to be compatible with each other so this function should be handled in `at_cmd_parser` library (or `nrf_modem_at.h`) so that the user of `nrf_modem_at` can just pass its output to `at_cmd_parser`. That was the case with `at_cmd` and `at_cmd_parser`.

**Comment by bjda** [comment]
> I think @bjda will submit a significant rewrite of this library fairly soon. Perhaps we should hold off until there's a clear timeline on that effort before merging this PR

These changes are pretty much orthogonal to mine for now, so this PR is fine from my perspective. I'll be adding new functions to read each parameter, change the parsing and introduce a more general "subscribe" for things like RSRP, battery, etc.

**Review by bjda — APPROVED** [review]

File: lib/modem_info/modem_info.c
Comment (lemrey) [reply lib/modem_info/modem_info.c:362]: I too think this should be handled by the parser, but if I made that change there now, it would not be compatible with libraries that still use `at_cmd` to receive the response. Once `at_cmd` is removed, we can make this change in the parser.

Comment (trantanen) [reply lib/modem_info/modem_info.c:362]: Yep, I know there would be a compatibility issue but thought it would be possible to accept both versions. Might be easier to do as you said but we need to get that `at_cmd_parser` fixed for 1.9.0 to avoid use of current functionality in any new code.

Comment (lemrey) [reply lib/modem_info/modem_info.c:362]: Alright, let me try and do this change in the parser and see how that looks.

Comment (lemrey) [reply lib/modem_info/modem_info.c:362]: @trantanen see: #6386

Comment (trantanen) [reply lib/modem_info/modem_info.c:362]: That PR looks good. I guess this PR should be marked with DNM label so that this function can be removed once 6386 is merged.

Comment (lemrey) [reply lib/modem_info/modem_info.c:362]: Added `DNM`, until #6386 is in and this has been rebased.

Comment (lemrey) [reply lib/modem_info/modem_info.c]: Rebased

**Review by trantanen — APPROVED** [review]

**Review by jtguggedal — CHANGES_REQUESTED** [review]
There seems to be issues with IP address parsing that needs to be resolved. Now that `OK\r\n` is included in the response it's interpreted as an IP address, which causes the parser function to return an error.

NACKing now just to avoid accidental merge while investigating

**Comment by lemrey** [comment]
Thanks @jtguggedal , I have updated the PR to trim the input buffer after the occurrence of `OK\r\n`, which should solve our issue.

File: lib/modem_info/modem_info.c
Comment (jtguggedal) [inline lib/modem_info/modem_info.c:477]: Neat solution!

**Review by jtguggedal — APPROVED** [review]
Thanks! Verified that the IP address parsing now works as expected

**Comment by lemrey** [comment]
Unrelated failure in some `matter` test, rebased to let CI re-run.

**Review by rlubos — APPROVED** [review]

**Pull request merged**
